VideoLocker/src/main/java/org/edx/mobile/view/CourseUnitWebviewFragment.java

```diff
@@ -71,6 +81,10 @@ public void onDestroy(){
         super.onDestroy();
     }
 
+    public void onEvent(NetworkConnectivityChangeEvent event){
+        setFragmentState(!NetworkUtil.isConnected(getContext()));
```

Comment:
I don't think we need to `registerSticky()` here, the current implementation handles the initial failure case kind of lazily i.e. we allow the webview to load the url and let it fail and when it does, we set the state accordingly using the `onReceivedError` callback.

If we proactively want to handle it, we can just apply a connectivity check webview initialisation code at the end of `onActivityCreated` function.
p.s we'll need the same check inside `setUserVisibleHint` function as well due to the *PRELOADING* feature implemented inside `ViewPagerDownloadManager` class.